src/controllers/restatedeployment/reconcilers/knative.rs

```diff
@@ -0,0 +1,938 @@
+        annotations.extend(user_annotations.clone());
+    }
+
+    // Step 2: Apply operator-managed autoscaling annotations (overriding user values)
+    if let Some(min) = knative_spec.min_scale {
+        annotations.insert("autoscaling.knative.dev/min-scale".to_string(), min.to_string());
+    } else {
+        // Default to scale-to-zero
+        annotations.insert("autoscaling.knative.dev/min-scale".to_string(), "0".to_string());
+    }
+
+    if let Some(max) = knative_spec.max_scale {
+        annotations.insert("autoscaling.knative.dev/max-scale".to_string(), max.to_string());
+    }
+
+    if let Some(target) = knative_spec.target {
+        annotations.insert("autoscaling.knative.dev/target".to_string(), target.to_string());
+    }
+
+    // Step 3: Always set parent deployment tracking (operator-managed)
+    annotations.insert(
+        RESTATE_DEPLOYMENT_ANNOTATION.to_string(),
+        rsd.name_any(),
+    );
+
+    // Get container spec from template
+    let containers = rsd
+        .spec
+        .template
+        .spec
+        .as_ref()
+        .and_then(|spec| spec.get("containers"))
+        .ok_or_else(|| Error::InvalidRestateDeployment("Missing containers in template".into()))?;
+
+    // Ensure Restate port and validate for Knative compatibility
+    let mut containers_array: Vec<serde_json::Value> = serde_json::from_value(containers.clone())?;
+    validate_knative_containers(&containers_array)?;
+    ensure_restate_port(&mut containers_array)?;
+    ensure_readiness_probe(&mut containers_array)?;
+
+    // Create owner reference
+    let owner_reference = rsd.controller_owner_ref(&()).unwrap();
+
+    Ok(json!({
+        "apiVersion": "serving.knative.dev/v1",
+        "kind": "Configuration",
+        "metadata": {
+            "name": name,
+            "namespace": namespace,
+            "ownerReferences": [owner_reference],
+            "labels": {
+                "app.kubernetes.io/managed-by": "restate-operator"
+            }
+        },
+        "spec": {
+            "template": {
+                "metadata": {
+                    "annotations": annotations,
+                    "labels": {
+                        "app.kubernetes.io/managed-by": "restate-operator"
+                    }
+                },
+                "spec": {
+                    "containers": containers_array,
+                }
+            }
+        }
+    }))
+}
+
+/// Validate containers for Knative compatibility
+/// Knative has strict requirements that differ from standard Kubernetes
+fn validate_knative_containers(containers: &[serde_json::Value]) -> Result<()> {
+    for (idx, container) in containers.iter().enumerate() {
+        // Validate port names
+        // Knative only allows port names: empty, "h2c", or "http1"
+        if let Some(ports) = container.get("ports").and_then(|p| p.as_array()) {
+            for port in ports {
+                if let Some(port_name) = port.get("name").and_then(|n| n.as_str()) {
```

Comment:
we can structure this as a match allowing None, h2c, and http1 but not allowing anythign else